scripts/bert/finetune_classifier.py

```diff
@@ -34,68 +34,99 @@
-parser.add_argument('--log_interval', type=int, default=10, help='report interval')
-parser.add_argument('--max_len', type=int, default=128, help='Maximum length of the sentence pairs')
+                    help='ratio of warmup steps used in NOAM\'s stepsize schedule, default is 0.1')
+parser.add_argument('--log_interval', type=int, default=10, help='report interval, default is 10')
+parser.add_argument('--max_len', type=int, default=128,
+                    help='Maximum length of the sentence pairs, default is 128')
+parser.add_argument('--seed', type=int, default=2, help='Random seed, default is 2')
+parser.add_argument('--accumulate', type=int, default=None, help='The number of batches for '
+                    'gradients accumulation to simulate large batch size. Default is None')
 parser.add_argument('--gpu', action='store_true', help='whether to use gpu for finetuning')
 args = parser.parse_args()
+
+logging.getLogger().setLevel(logging.DEBUG)
 logging.info(args)
+
 batch_size = args.batch_size
-test_batch_size = args.test_batch_size
+dev_batch_size = args.dev_batch_size
 lr = args.lr
+accumulate = args.accumulate
+log_interval = args.log_interval * accumulate if accumulate else args.log_interval
+
+# random seed
+np.random.seed(args.seed)
+random.seed(args.seed)
+mx.random.seed(args.seed)
 
 ctx = mx.cpu() if not args.gpu else mx.gpu()
 
+# model and loss
 dataset = 'book_corpus_wiki_en_uncased'
 bert, vocabulary = bert_12_768_12(dataset_name=dataset,
                                   pretrained=True, ctx=ctx, use_pooler=True,
                                   use_decoder=False, use_classifier=False)
-do_lower_case = 'uncased' in dataset
-tokenizer = FullTokenizer(vocabulary, do_lower_case=do_lower_case)
-
 model = BERTClassifier(bert, dropout=0.1)
 model.classifier.initialize(init=mx.init.Normal(0.02), ctx=ctx)
-logging.info(model)
 model.hybridize(static_alloc=True)
 
 loss_function = gluon.loss.SoftmaxCELoss()
 loss_function.hybridize(static_alloc=True)
-
 metric = mx.metric.Accuracy()
 
-trans = ClassificationTransform(tokenizer, MRPCDataset.get_labels(), args.max_len)
-data_train = MRPCDataset('train').transform(trans)
-data_dev = MRPCDataset('dev').transform(trans)
-
-bert_dataloader = mx.gluon.data.DataLoader(data_train, batch_size=batch_size,
-                                           shuffle=True, last_batch='rollover')
-bert_dataloader_dev = mx.gluon.data.DataLoader(data_dev, batch_size=test_batch_size,
-                                               shuffle=False)
+# data processing
+do_lower_case = 'uncased' in dataset
+tokenizer = FullTokenizer(vocabulary, do_lower_case=do_lower_case)
+train_trans = ClassificationTransform(tokenizer, MRPCDataset.get_labels(),
+                                      args.max_len, pad=False)
+dev_trans = ClassificationTransform(tokenizer, MRPCDataset.get_labels(), args.max_len)
+
+data_train = MRPCDataset('train').transform(train_trans, lazy=False)
```

Comment:
consider modularizing the data pipeline by making some parts into a function. this helps avoid leaking variables into global context.